### PR TITLE
RIA-7908 change_party_to_appelant

### PR DESCRIPTION
**Pull request by ikirsanov**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RIA-7908

### Change description ###
Changed the party to Appellant conditions in standard directions for Internal case events:

- (Request case building - both ADA and non ADA)
-  (Review Home Office response - Non ADA only)
- (Request hearing requirement - Non ADA only)
- Added new junit tests and edited existed

### Checklist

- [ x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestHearingRequirementsDirectionHandler.java [hunk 1]
```diff
@@ -90,4 +91,11 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
 
         return new PreSubmitCallbackResponse<>(asylumCase);
     }
+
+    private Parties resolvePartiesForHearingRequirements(AsylumCase asylumCase) {
+        if (HandlerUtils.isAipJourney(asylumCase) || isInternalCase(asylumCase) && isAppellantInDetention(asylumCase) && !isAcceleratedDetainedAppeal(asylumCase)) {
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestHearingRequirementsDirectionHandler.java:96]: (HandlerUtils.isAipJourney(asylumCase) || 
(isInternalCase(asylumCase) && isAppellantInDetention(asylumCase) && !isAcceleratedDetainedAppeal(asylumCase))) 
Can we please change the condition.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestHearingRequirementsDirectionTest.java [hunk 2]
```diff
@@ -96,8 +94,8 @@ void can_handle_request_hearing_requirements_feature() {
     }
 
     @ParameterizedTest
-    @EnumSource(value = Parties.class, names = {"LEGAL_REPRESENTATIVE", "APPELLANT"})
-    void should_append_new_direction_to_existing_directions_for_the_case(Parties party) {
+    @MethodSource("caseTypeScenarios")
+    void should_append_new_direction_to_existing_directions_for_the_case(YesOrNo yesOrNo, YesOrNo ada, JourneyType journeyType, Parties party) {
```
Comment (neha-aggarw) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestHearingRequirementsDirectionTest.java:98]: Please rename args to be more meaningful - like YesOrNo isAppellantInDetention

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestResponseReviewHandlerTest.java [hunk 3]
```diff
@@ -58,18 +69,21 @@ public void setUp() {
             new RequestResponseReviewHandler(DUE_IN_DAYS, dateProvider);
     }
 
-    @Test
-    void should_prepare_send_direction_fields() {
+    @ParameterizedTest
+    @MethodSource("caseTypeScenarios")
+    void should_prepare_send_direction_fields(YesOrNo yesOrNo, YesOrNo ada, Parties expectedParties) {
```
Comment (neha-aggarw) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestResponseReviewHandlerTest.java:74]: same as above.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/service/DirectionPartiesResolverTest.java [hunk 4]
```diff
@@ -39,27 +49,24 @@ public void setUp() {
         directionPartiesResolver = new DirectionPartiesResolver();
     }
 
-    @Test
-    void should_return_parties_for_send_direction_events() {
-
+    @ParameterizedTest
+    @MethodSource("caseTypeScenarios")
+    void should_return_parties_for_send_direction_events(YesOrNo yesOrNo, Parties parties) {
```
Comment (neha-aggarw) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/service/DirectionPartiesResolverTest.java:54]: same as above.

**Review by neha-aggarw — CHANGES_REQUESTED** [review]
The description of Fts should mention if the scenario is of Internal appeal (along with detention information).

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Review by neha-aggarw — APPROVED** [review]

**Pull request merged**